### PR TITLE
Introduce `BasicBlock::KindIs()` for checking bbJumpKind

### PR DESCRIPTION
**Pull request by BruceForstall**
This can simplify code that is checking for multiple `bbJumpKind` kinds with
an `if` statement, where a `switch` is not warranted. It's the same
usage model as `GenTree::OperIs()`.

e.g.,
```
if (block->KindIs(BBJ_COND, BBJ_SWITCH, BBJ_RETURN))
```

### CODE REVIEW

**Comment by BruceForstall** [comment]
What do people think?

I converted a bunch of cases where more than one `bbJumpKind` was being checked in an `if`; I like the simplicity that results.

It does mean that we could end up with a mix of `block->KindIs(BBJ_NONE)` and `(block->bbJumpKind == BBJ_NONE)` code style in the code if people start using `KindIs` for single cases, but I suppose we already have that for `OperIs` currently.

File: src/coreclr/jit/block.h [hunk 1]
```diff
@@ -829,6 +829,17 @@ struct BasicBlock : private LIR::Range
         BBswtDesc*  bbJumpSwt;  // switch descriptor
     };
 
+    bool KindIs(BBjumpKinds oper) const
```
Comment (EgorBo) [inline src/coreclr/jit/block.h:832]: I'd rename `oper` to `kind` or `jumpKind`

**Review by EgorBo — APPROVED** [review]

File: src/coreclr/jit/fgopt.cpp [hunk 2]
```diff
@@ -145,16 +145,16 @@ bool Compiler::fgReachable(BasicBlock* b1, BasicBlock* b2)
 
     if (b1->bbNum > fgDomBBcount)
     {
-        noway_assert(b1->bbJumpKind == BBJ_NONE || b1->bbJumpKind == BBJ_ALWAYS || b1->bbJumpKind == BBJ_COND);
+        noway_assert(b1->KindIs(BBJ_NONE, BBJ_ALWAYS, BBJ_COND));
 
-        if (b1->bbFallsThrough() && fgReachable(b1->bbNext, b2))
+        if (b1->KindIs(BBJ_NONE, BBJ_COND) && fgReachable(b1->bbNext, b2))
```
Comment (jakobbotsch) [inline src/coreclr/jit/fgopt.cpp:150]: Doesn't this change behavior? `bbFallsThrough()` checks a flag for `BBJ_CALLFINALLY`.

File: src/coreclr/jit/fgopt.cpp [hunk 3]
```diff
@@ -145,16 +145,16 @@ bool Compiler::fgReachable(BasicBlock* b1, BasicBlock* b2)
...
         {
             return true;
         }
 
-        if (b1->bbJumpKind == BBJ_ALWAYS || b1->bbJumpKind == BBJ_COND)
+        if (b1->KindIs(BBJ_ALWAYS, BBJ_COND) && fgReachable(b1->bbJumpDest, b2))
         {
-            return fgReachable(b1->bbJumpDest, b2);
+            return true;
         }
```
Comment (jakobbotsch) [inline src/coreclr/jit/fgopt.cpp:158]: I'm curious if this recursive call is still eliminated after this change. Not sure if it matters.

**Review by jakobbotsch — COMMENTED** [review]
LGTM otherwise.

File: src/coreclr/jit/fgopt.cpp (hunk 2 above)
Comment (BruceForstall) [reply src/coreclr/jit/fgopt.cpp:150]: We already `noway_assert` for it being NONE/ALWAYS/COND, so we won't see a BBJ_CALLFINALLY here.

File: src/coreclr/jit/fgopt.cpp (hunk 3 above)
Comment (BruceForstall) [reply src/coreclr/jit/fgopt.cpp:158]: Interesting point, but IMO, it doesn't matter, and the increased regularity/symmetry is preferable.

**Pull request merged**